### PR TITLE
DoH

### PR DESCRIPTION
**Pull request by james-otten**
Make the DoH server on the recursive resolver have a valid cert via Let's Encrypt + certbot + certbot-dns-rfc2136. This creates and removes records on a specific authoritative server. Authentication for this is via TSIG.

- [x] Set new secret in both environments

### CODE REVIEW

File: infra/ansible/roles/knot-authoritative/templates/knot.conf.j2 [hunk 1]
```diff
@@ -25,10 +25,23 @@ template:
     global-module: mod-stats
     storage: /var/lib/knot/zones
 
+key:
+  - id: doh.mesh.nycmesh.net
+    algorithm: hmac-sha384
```
Comment (Andrew-Dickinson) [inline infra/ansible/roles/knot-authoritative/templates/knot.conf.j2:30]: 512?

File: infra/ansible/roles/knot-authoritative/templates/knot.conf.j2 [hunk 2]
```diff
@@ -25,10 +25,23 @@ template:
...
+    secret: {{ TSIG_KEY_DOH }}
+
+acl:
+  - id: acl-update-doh
+    action: update
+    key: doh.mesh.nycmesh.net
```
Comment (Andrew-Dickinson) [inline infra/ansible/roles/knot-authoritative/templates/knot.conf.j2:36]: Restrict to `acme-challenge.doh.mesh.nycmesh.net. txt`?

**Review by Andrew-Dickinson — APPROVED** [review]

File: infra/ansible/roles/knot-authoritative/templates/knot.conf.j2 (hunk 1 above)
Comment (james-otten) [reply infra/ansible/roles/knot-authoritative/templates/knot.conf.j2]: Good call, it supports it. Regenerated and updated the algo

File: infra/ansible/roles/knot-authoritative/templates/knot.conf.j2 (hunk 2 above)
Comment (james-otten) [reply infra/ansible/roles/knot-authoritative/templates/knot.conf.j2]: Good call. Limited to a specific txt record,

**Pull request merged**